script.js

```diff
@@ -0,0 +1,621 @@
+};
+// returns true if there is a flush in the player's hand
+const isFlush = () => {
+  // if checkFlush is true, there is a Flush in the player's hand
+  let checkFlush = false;
+
+  // logic
+  if (playerHand[0].suit === playerHand[1].suit && playerHand[1].suit === playerHand[2].suit
+    && playerHand[2].suit === playerHand[3].suit && playerHand[3].suit === playerHand[4].suit) {
+    checkFlush = true;
+  }
+
+  return checkFlush;
+};
+// return true if there is a royal flush (straight from 10 to ace and cards have same suit)
+const isRoyalFlush = () => {
+  // if check10ToAce is true, there is a straight from 10 to Ace in the player's hand
+  let check10ToAce = false;
+  // number of times the difference between playerHand[i].rank and playerHand[i+1].rank is one
+  let timesDifferenceIsMinusOne = 0;
+  // if there is a flush, check for straight from 10 to ace
+  if (isFlush() === true) {
+    // check if player has a straight from 10 to ace
+    for (let i = 0; i < (playerHand.length - 2); i += 1) {
+      if (playerHand[i].rank - playerHand[i + 1].rank === 1) {
+        timesDifferenceIsMinusOne += 1;
+      }
+    }
+    if (timesDifferenceIsMinusOne === 3 && playerHand[4].rank === 1 && playerHand[0].rank === 13) {
+      check10ToAce = true;
+    }
+  }
+
+  return check10ToAce;
+};
+// returns true if there is a full house in the player's hand
+const isFullHouse = () => {
+  let checkFullHouse = false;
+  if (numOf3OfAKind === 1 && numOfPairs === 1) {
+    checkFullHouse = true;
+  }
+  return checkFullHouse;
+};
+// returns true if there is a card that is Jack or higher in the player's hand
+const isJackOrHigher = () => {
+  let checkJackOrHigher = false;
+  // check every card in player's hand for jack,queen,king or ace
+  for (let i = 0; i < playerHand.length; i += 1) {
+    if (playerHand[i].rank > 10 || playerHand[i].rank === 1) {
+      checkJackOrHigher = true;
+    }
+  }
+  return checkJackOrHigher;
+};
+// returns number of points based on player's hand
+const calcHandScore = () => {
+  if (isRoyalFlush() === true) { // royal flush
+    handScore = 10;
+  } else if (isStraight() === true && isFlush() === true) { // straight flush
+    handScore = 9;
+  } else if (numOf4OfAKind === 1) { // 4 of a kind
+    handScore = 8;
+  } else if (isFullHouse() === true) { // full house
+    handScore = 7;
+  } else if (isFlush() === true) { // flush
+    handScore = 6;
+  } else if (isStraight() === true) { // straight
+    handScore = 5;
+  } else if (numOf3OfAKind === 1) { // 3 of a kind
+    handScore = 4;
+  } else if (numOfPairs === 2) { // 2 pairs
+    handScore = 3;
+  } else if (numOfPairs === 1) { // 1 pair
+    handScore = 2;
+  } else if (isJackOrHigher() === true) {
+    handScore = 1;
+  } else {
+    handScore = 0;
+  }
+};
+
+// returns points won for the round
+const calcPointsWon = () => playerBidPoints * handScore;
+// add points to player's total points based on bid points and hand score
+const addPoints = () => {
```

Comment:
This function name is ambiguous and could be more specific. Add what points to what?